### PR TITLE
This PR contains all new buildings and some fixes from the more buildings PR

### PR DESCRIPTION
**Pull request by andreas-eberle**
This PR only contains UI stuff and minor logic fixes. The game logic for those buildings is not included.

Included buildings:
- hospital 
- donkey farm
- marketplace
- habor

reworked UI panels: 
- marketplace panel

### CODE REVIEW

File: jsettlers.graphics/src/jsettlers/graphics/localization/Labels.java [hunk 1]
```diff
@@ -131,4 +131,9 @@ public static String getName(EActionType action) {
 	public static String getProgress(EProgressState loading) {
 		return getString("progress_" + loading);
 	}
+
+	public static String getName(EMaterialType material) {
+		// TODO Auto-generated method stub
```
Comment (codingberlin) [inline jsettlers.graphics/src/jsettlers/graphics/localization/Labels.java]: Is this TODO for the PR or for later implementation?

**Comment by codingberlin** [comment]
I tested the PR.
When creating a pier:
```
Error during execution of scheduled task:
java.lang.NullPointerException
	at jsettlers.logic.map.grid.MainGrid.constructBuildingAt(MainGrid.java:387)
	at jsettlers.logic.map.grid.MainGrid$GuiInputGrid.constructBuildingAt(MainGrid.java:1761)
	at jsettlers.input.GuiTaskExecutor.executeTask(GuiTaskExecutor.java:89)
	at jsettlers.network.synchronic.timer.NetworkTimer.executeTasksPacket(NetworkTimer.java:168)
	at jsettlers.network.synchronic.timer.NetworkTimer.executeRun(NetworkTimer.java:141)
	at jsettlers.network.synchronic.timer.NetworkTimer.run(NetworkTimer.java:108)
	at java.util.TimerThread.mainLoop(Timer.java:555)
	at java.util.TimerThread.run(Timer.java:505)
```

File: jsettlers.graphics/src/jsettlers/graphics/localization/Labels.java (hunk 1 above)
Comment (michaelzangl) [inline jsettlers.graphics/src/jsettlers/graphics/localization/Labels.java]: Anytime you want. This function is already included in the map editor. You need to move it from the map editor to graphics (and move the labels)

**Comment by codingberlin** [comment]
stones and planks are duplicated in the market menu
bildschirmfoto 2015-11-13 um 13 11 02

**Comment by michaelzangl** [comment]
Known bug: Those materials are not added to EMaterialType, so stone and plank are placeholders.

**Comment by codingberlin** [comment]
bildschirmfoto 2015-11-13 um 13 12 01
pier or harbor have a wrong building (ether in the menu or in the game)

**Comment by codingberlin** [comment]
goods were not transported to the market
bildschirmfoto 2015-11-13 um 13 13 26

**Comment by codingberlin** [comment]
The harbor has the yard building and translation is missing (building_HARBOR).
bildschirmfoto 2015-11-13 um 13 15 37

**Comment by andreas-eberle** [comment]
The dockyard has a wrong preview image. I tryed to find the correct preview image index with the DatFileTester but I was not able to see it. When I scroll in the list of images, they scroll out of the screen, so I cannot see them. @michaelzangl: Am I doing something wrong?

**Comment by codingberlin** [comment]
@andreas-eberle for dockyard (I experienced the same bug with the DatFileTester. I searched then in the folders):
```
<image file="14" for="GUI" type="GUI" sequence=108" image="0"/>
<image file="13" for="FINAL" type="SETTLER" sequence="44" image="0"/>
<image file="13" for="BUILD" type="SETTLER" sequence="44" image="1"/>
```

**Comment by michaelzangl** [comment]
The DatFileTester is a bit broken. Simply export the images and search the folders.

**Comment by michaelzangl** [comment]
@codingberlin The harbour/market in this PR is not working, the logic is missing.

**Comment by codingberlin** [comment]
My fault. I didn't recognized the ```The game logic for those buildings is not included.``` Sorry

**Comment by andreas-eberle** [comment]
@codingberlin: The exception you posted is fixed.

File: jsettlers.graphics/src/jsettlers/graphics/localization/Labels.java (hunk 1 above)
Comment (andreas-eberle) [inline jsettlers.graphics/src/jsettlers/graphics/localization/Labels.java]: Wouldn't it be possible to just use https://github.com/jsettlers/settlers-remake/pull/236/files#diff-ab8ae3ce41667a0e8e2a3d65bb8974f8R105 ?

**Pull request merged**